File: solutions/cpp/log-levels/1/log_levels.cpp
#include <string>
#define String std::string

namespace log_line {
    String message(String line) {
        int space_index = line.find(" ");
        return line.substr(space_index + 1);
    }

    String log_level(String line) {
        int first_bracket = line.find("[");
        int last_bracket = line.find("]");
        return line.substr(first_bracket +1 ,last_bracket -1);
    }

    String reformat(String line) {
        String msg = message(line);
        String lvl = log_level(line);
        return msg + " (" + lvl + ")";
    }
}

File: solutions/clojure/log-levels/1/src/log_levels.clj
(ns log-levels
  (:require [clojure.string :as str]))

(defn message
  [s]
  (str/trim (last (str/split s #":")))
)

(defn log-level
  [s]
  (-> (re-find #"\w+" s) ;
      str/lower-case))

(defn reformat [s]
  (->> (str/split s #":")
       (map str/trim)
       ((fn [[level msg]]
          (format "%s (%s)" msg (str/lower-case (str/replace level #"[\[\]]" "")))))
       ))